File: check_if_it_Is_a_straight_line.py
class Solution:
    def checkStraightLine(self, coordinates: List[List[int]]) -> bool:
        x0, y0 = coordinates[0]
        x1, y1 = coordinates[1]

        for i in range(2, len(coordinates)):
            x, y = coordinates[i]

            if (y1 - y0) * (x - x0) != (y - y0) * (x1 - x0):
                return False

        return True        

File: check_if_it_Is_a_straight_line.js
/**
 * @param {number[][]} coordinates
 * @return {boolean}
 */
var checkStraightLine = function(coordinates) {
    const [x0, y0] = coordinates[0];
    const [x1, y1] = coordinates[1];

    for (let i = 2; i < coordinates.length; i++) {
        const [x, y] = coordinates[i];

        if ((y1 - y0) * (x - x0) !== (y - y0) * (x1 - x0)) {
            return false;
        }
    }

    return true;    
};
